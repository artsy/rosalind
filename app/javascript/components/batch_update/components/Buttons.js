import styled, { css } from 'styled-components'
import { avantGarde, colors } from './Layout'

const buttonSizing = (props) => {
  if (props.fullWidth) {
    return css`
      display: block;
      width: 100%;
      padding: 18px 30px 15px 30px;
    `
  } else {
    return css`
      display: inline-block;
      padding: 18px 30px 15px 30px;
    `
  }
}

const buttonColors = (props) => {
  if (props.primary) {
    return css`
      background: ${colors.black};
      color: ${colors.white};
      border: solid 1px ${colors.black};
    `
  } else if (props.secondary) {
    return css`
      background: ${colors.white};
      color: ${colors.black};
      border: solid 1px ${colors.black};
    `
  } else if (props.disabled) {
    return css`
      background: ${colors.grayLightest};
      color: ${colors.gray};
      border: solid 1px ${colors.grayLightest};
    `
  } else {
    return css`
      background: ${colors.grayLightest};
      color: ${colors.black};
      border: solid 1px ${colors.grayLightest};
    `
  }
}

const buttonTypography = (props) => {
  return css`
    ${avantGarde}
    font-weight: normal;
    font-size: 13px;
    line-height: 13px;
    text-transform: uppercase;
  `
}

const buttonTransitions = (props) => {
  if (props.disabled) {
    return css`
      transition: none;
    `
  } else {
    return css`
      transition-property: background, border-color, color;
      transition-duration: 0.15s;
      &:hover {
        background: ${colors.purple};
        border-color: ${colors.purple};
        color: ${colors.white};
      }
    `
  }
}

const buttonCursor = (props) => {
  if (props.disabled) {
    return css`
      cursor: not-allowed;
    `
  } else {
    return css`
      cursor: pointer;
    `
  }
}

export const Button = styled.button`
  ${buttonSizing}
  ${buttonColors}
  ${buttonTypography}
  ${buttonTransitions}
  ${buttonCursor}
`

export const LinkButton = styled.a`
  ${buttonSizing}
  ${buttonColors}
  ${buttonTypography}
  ${buttonTransitions}
  ${buttonCursor}

  text-decoration: none; 
  text-align: center;
`
