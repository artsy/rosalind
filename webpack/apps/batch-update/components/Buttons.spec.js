import React from 'react'
import renderer from 'react-test-renderer'
import { Button, LinkButton } from './Buttons'
import { mount } from 'enzyme'

describe('Button', () => {
  it('renders a basic button', () => {
    const rendered = renderer.create(<Button>Click me</Button>)
    const tree = rendered.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders a primary button', () => {
    const rendered = renderer.create(<Button primary>Click me</Button>)
    const tree = rendered.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders a secondary button', () => {
    const rendered = renderer.create(<Button secondary>Click me</Button>)
    const tree = rendered.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders a full width button', () => {
    const rendered = renderer.create(<Button fullWidth>Click me</Button>)
    const tree = rendered.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders a full width button', () => {
    const rendered = renderer.create(<LinkButton fullWidth>Click me</LinkButton>)
    const tree = rendered.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('dispatches the click handler on a normal button', () => {
    const clickHandler = jest.fn()
    const wrapper = mount(<Button onClick={clickHandler} />)
    wrapper.find('button').simulate('click')
    expect(clickHandler.mock.calls.length).toEqual(1)
  })

  describe('when "disabled" prop is supplied', () => {
    it('renders a disabled button', () => {
      const rendered = renderer.create(<Button disabled>Click me</Button>)
      const tree = rendered.toJSON()
      expect(tree).toMatchSnapshot()
    })

    it('does not dispatch the click handler', () => {
      const clickHandler = jest.fn()
      const wrapper = mount(<Button disabled onClick={clickHandler} />)
      wrapper.find('button').simulate('click')
      expect(clickHandler.mock.calls.length).toEqual(0)
    })
  })
})

describe('LinkButton', () => {
  it('renders a basic link button', () => {
    const rendered = renderer.create(<LinkButton>Click me</LinkButton>)
    const tree = rendered.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders a primary link button', () => {
    const rendered = renderer.create(<LinkButton primary>Click me</LinkButton>)
    const tree = rendered.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders a secondary link button', () => {
    const rendered = renderer.create(<LinkButton secondary>Click me</LinkButton>)
    const tree = rendered.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders a full width link button', () => {
    const rendered = renderer.create(<LinkButton fullWidth>Click me</LinkButton>)
    const tree = rendered.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders a full width link button', () => {
    const rendered = renderer.create(<LinkButton fullWidth>Click me</LinkButton>)
    const tree = rendered.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders a disabled link button', () => {
    const rendered = renderer.create(<LinkButton disabled>Click me</LinkButton>)
    const tree = rendered.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
