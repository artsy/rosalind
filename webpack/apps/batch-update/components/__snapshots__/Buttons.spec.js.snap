exports[`Button renders a basic button 1`] = `
<button
  className="cPyjKp">
  Click me
</button>
`;

exports[`Button renders a full width button 1`] = `
<button
  className="kTrUEX">
  Click me
</button>
`;

exports[`Button renders a full width button 2`] = `
<a
  className="jpUmnI">
  Click me
</a>
`;

exports[`Button renders a primary button 1`] = `
<button
  className="hwoZit">
  Click me
</button>
`;

exports[`Button renders a secondary button 1`] = `
<button
  className="hggFYC">
  Click me
</button>
`;

exports[`Button when "disabled" prop is supplied renders a disabled button 1`] = `
<button
  className="feplCa"
  disabled={true}>
  Click me
</button>
`;

exports[`LinkButton renders a basic link button 1`] = `
<a
  className="bMOQzN">
  Click me
</a>
`;

exports[`LinkButton renders a disabled link button 1`] = `
<a
  className="bqDpOh"
  disabled={true}>
  Click me
</a>
`;

exports[`LinkButton renders a full width link button 1`] = `
<a
  className="jpUmnI">
  Click me
</a>
`;

exports[`LinkButton renders a full width link button 2`] = `
<a
  className="jpUmnI">
  Click me
</a>
`;

exports[`LinkButton renders a primary link button 1`] = `
<a
  className="jhxkip">
  Click me
</a>
`;

exports[`LinkButton renders a secondary link button 1`] = `
<a
  className="jRkNFJ">
  Click me
</a>
`;
